### PR TITLE
Suppression list Insert and Delete

### PR DESCRIPTION
**Pull request by asgoel**
@ewandennis added ability to insert/update or delete a single suppression list entry.

Also had to add one more error check to the endpoint response parser to handle normal HTTPoison errors (such as request timeouts).

### CODE REVIEW

File: lib/suppression_list.ex [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
     - %SparkPost.SuppressionList.SearchResult{}
   """
 
   alias SparkPost.Endpoint
 
   @doc """
+  Insert or update a single entry in the suppression list.
+  Returns a single string with the success message if the entry
+  was updated or inserted. Returns a %SparkPost.Endpoint.Error{} with a 400
+  if the type is invalid.
```
Comment (ewandennis) [inline lib/suppression_list.ex]: "...if there was an issue with the request format" ?

File: lib/suppression_list.ex [hunk 2]
```diff
@@ -1,18 +1,69 @@
...
+
+  Parameters:
+    - recipient: the email to insert or update in the suppression list
+    - type: one of "transactional" or "non_transactional"
+    - description (optional): optional description of this entry in the suppression list
+  """
+  def update_one(recipient, type, description \\ nil) do
```
Comment (ewandennis) [inline lib/suppression_list.ex]: Think we should call it `insert_or_update_one/3` or `upsert_one/3`?

File: lib/suppression_list.ex [hunk 3]
```diff
@@ -1,18 +1,69 @@
...
+    body = if description == nil do
+      %{type: type}
+    else
+      %{type: type, description: description}
+    end
+    response = Endpoint.request(:put, "suppression-list/#{recipient}", body)
```
Comment (ewandennis) [inline lib/suppression_list.ex]: I wonder if we should adopt the usual `{:ok, %Sparkpost.Endpoint.Response{}}` or `{:error, %SparkPost.Endpoint.Error{}}` here.

File: lib/suppression_list.ex [hunk 4]
```diff
@@ -1,18 +1,69 @@
...
+  @doc """
+  Deletes a specific entry from the list. Returns an empty string if
+  the deletion was successful. Returns a %SparkPost.Endpoint.Error{} with a 404
+  if the specified entry is not in the list. Returns a %SparkPost.Endpoint.Error{}
+  with a 403 if the entry could not be removed for any reason (such as Compliance).
+
+  Parameters:
+    recipient: the entry to delete from the suppression list.
+  """
+  def delete(recipient) do
```
Comment (ewandennis) [inline lib/suppression_list.ex]: The same comment about idiomatic `{:ok, ...}` return values applies here.

File: lib/endpoint.ex [hunk 5]
```diff
@@ -73,9 +73,9 @@ defmodule SparkPost.Endpoint do
   defp handle_response({:ok, %HTTPoison.Response{status_code: code, body: body}}, decode_results) when code >= 200 and code < 300 do
     decoded_body = decode_response_body(body)
     if decode_results do
-      %SparkPost.Endpoint.Response{status_code: 200, results: decoded_body.results}
+      %SparkPost.Endpoint.Response{status_code: code, results: decoded_body.results}
```
Comment (ewandennis) [inline lib/endpoint.ex]: 👍

Comment (ewandennis) [inline lib/endpoint.ex]: Will have to check which other modules either make assumptions about the status code or pass it on to client code. This could produce a (good) breaking change in those cases.

File: lib/suppression_list.ex (hunk 3 above)
Comment (asgoel) [inline lib/suppression_list.ex]: I'd like to adopt this. Unfortunately I'm not sure I have enough time to go through and update the library everywhere to make sure everything still works. Might make sense to have this in a different PR?

Comment (ewandennis) [inline lib/suppression_list.ex]: Yup. I wasn't suggesting you retrofit the whole library with this. These are the first endpoints with simple good/bad responses though so I thought we'd start here.

File: lib/endpoint.ex (hunk 5 above)
Comment (asgoel) [inline lib/endpoint.ex]: didn't see anything on a cursory glance. Let me know if I missed anything though.

File: lib/suppression_list.ex (hunk 2 above)
Comment (asgoel) [inline lib/suppression_list.ex]: `upsert_one` sounds good

File: lib/suppression_list.ex (hunk 1 above)
Comment (asgoel) [inline lib/suppression_list.ex]: sure

**Comment by ewandennis** [comment]
Sorry for the delay on this. I'm going to merge, do a little cleanup elsewhere and cut a release. Thanks as ever @asgoel !
